client/src/view/actions-panel.tsx

```diff
@@ -30,7 +37,7 @@ export class ActionsPanel extends React.Component<IActionsPanelProps, object> {
     }
 
     public componentDidMount(): void {
-        this.subs.push(this.props.islandState.subject.subscribe(islandState => this.setState({islandState})))
+        this.subs.push(this.props.stateSubject.subscribe(appState => this.setState({appState})))
```

Comment:
I see you push to a `subs` array in a couple of places to do mass unsubscribes. Is there no function like `stateSubject.unsubscribeAll()`? Seems like that should be a feature

Reply:
Should be I guess.  But you're never sure when mount/unmount will happen for individual components.

Reply:
Do multiple components subscribe to the same subject? In that case this approach is fine